cachito/workers/pkg_managers/general_js.py

```diff
@@ -44,6 +51,67 @@
 log = logging.getLogger(__name__)
 
 
+def parse_dependency(proxy_repo_url, dep_identifier):
+    """
+    Parse the dependecy to return the proxied download URL and the file name.
+
+    :param str proxy_repo_url: The Nexus proxy repository URL to use as the registry
+    :param str dep_identifier:
+    :return: proxied_url for the download and the tarball_name.
+    :rtype: str, str
+    """
+    full_pkg_name, version = dep_identifier.rsplit("@", 1)
+    if "/" in full_pkg_name:
+        short_pkg_name = full_pkg_name.rsplit("/", 1)[1]
+    else:
+        short_pkg_name = full_pkg_name
+
+    proxied_url = f"{proxy_repo_url}{full_pkg_name}/-/{short_pkg_name}-{version}.tgz"
+    tarball_name = f"{full_pkg_name}-{version}.tgz".replace("@", "").replace("/", "-")
```

Comment:
How did you determine the expected format of the url and the downloaded tarball name?

Reply:
https://docs.google.com/document/d/1vG5_GlS_P5f7m87ezqtEnXTwv87rn7jy7yZk6LBU7HA/edit#heading=h.suij2g8c5kf8

It was present in the design document.

Reply:
While diving through the npm CLI, I also found the relevant code for the filename: https://github.com/npm/cli/blob/84598fe2b84d57b812f53ecc0beba9a04e75c069/workspaces/libnpmpack/lib/index.js#L41

One minor detail is that the '@' should only be replaced if it's at the beginning of the package name, otherwise this seems correct :+1:

Reply:
Well observed, Adam! I changed it to a regex that checks if the '@' is at the beginning.